caddyconfig/httpcaddyfile/httptype.go

```diff
@@ -133,14 +143,17 @@ func (st ServerType) Setup(originalServerBlocks []caddyfile.ServerBlock,
 				return nil, warnings, fmt.Errorf("%s:%d: unrecognized directive: %s", tkn.File, tkn.Line, dir)
 			}
 
-			results, err := dirFunc(Helper{
+			h := Helper{
 				Dispenser:    caddyfile.NewDispenser(segment),
 				options:      options,
 				warnings:     &warnings,
 				matcherDefs:  matcherDefs,
 				parentBlock:  sb.block,
 				groupCounter: gc,
-			})
+				state:        state,
```

Comment:
We can just initialize state right here
```suggestion
				state:        make(map[string]interface{}),
```

Reply:
Hmm i'm not so sure about that.

Reply:
Oh wait. You are right